### PR TITLE
feat: add a flake module for use via flake.parts

### LINKED ISSUE #238: Expose `flakeModule` from flake for use with `flake-parts`
Is there any interest in providing a `flake-parts` module? @roberth had a WIP solution here: https://github.com/numtide/devshell/compare/master...hercules-ci:devshell:flake-modules

I'm happy to resume/update this if there is any interest.

[Matrix discussion](https://matrix.to/#/!gcrYWdPsIUOFpXFDHB:matrix.org/$f6D30PYnTtYwXgCSkcL7Vnw-xTg1-EMTfM1OZ8BycX8?via=matrix.org&via=ukvly.org&via=valkor.net)

My intention is to use devshell's MOTD feature but without having to create devshell.toml and the like (ie., configure everything in flake.nix, and use with an existing language dev shell). To begin with, I'd use this to replace all the scripts in https://github.com/srid/haskell-template/tree/master/bin

### PR DESCRIPTION
**Pull request by brianmcgee**
Based on the suggestion from @terlar 

Closes #238

Allows users to use devshell within a [flake.parts](https://flake.parts) based flake like so:

```nix
{inputs, ...}: {
	imports = [
		inputs.devshell.flakeModule
	]; 

	perSystem = {
		devshell.shells.default = {
			env = [ ];
			commands = [ ];
			packages = [ ];
		};

		devshell.shells.projectA = {
			env = [ ];
			commands = [ ];
			packages = [ ];
		};

		devshell.shells.projectB = {
			env = [ ];
			commands = [ ];
			packages = [ ];
		};
	};
}

```

### CODE REVIEW

File: flake-module.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      pkgs,
+      ...
+    }: let
+      devshellType =
+        (lib.evalModules {
+          modules = import ./modules/modules.nix {
+            inherit lib pkgs;
+          };
+        })
+        .type;
```
[the remark this answers is not in the document]
Comment (brianmcgee) [reply flake-module.nix:20]: I don't follow, can you elaborate?

File: flake-module.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    in {
+      options.devshell.shells = with lib; mkOption {
```
[the remark this answers is not in the document]
Comment (brianmcgee) [reply flake-module.nix]: This has been done.

[the remark this answers is not in the document]
Comment (brianmcgee) [reply flake-module.nix]: This has been done

[the remark this answers is not in the document]
Comment (brianmcgee) [reply flake-module.nix]: I know what you mean, but I also can't think of a good alternative.

File: flake.nix [hunk 3]
```diff
@@ -27,12 +27,23 @@
...
+      templates = rec {
+        toml = {
+          path = ./templates/toml;
+          description = "nix flake new my-project -t github:numtide/devshell";
+        };
+        flake-parts = {
+          path = ./templates/flake-parts;
+          description = "nix flake new my-project -t github:numtide/devshell#flake-parts";
+        };
+        default = toml;
```
Comment (brianmcgee) [inline flake.nix:40]: @zimbatm I added a template as well, keeping the original toml based one as the default.

File: flake-module.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (zimbatm) [reply flake-module.nix]: Let's add a "not to be confused with `devShell`, with a capital S. We know it's unfortunate, get used to it" to the description :)

**Comment by roberth** [comment]
Rendered docs: https://flake.parts/options/devshell.html

**Pull request merged**